include/ restructring and header cleanup

nashif opened the pull request:
This is a major change folloing TSC F2F discussion:

- Moved misc/* to sys/*
- Moved all peripheral APIs into drivers/
- Moved headers into directory matching subsystem they provide APIs for
- Other cleanup of private and internal APIs

All related to #16539

aescolar reviewed (dismissed):
The TSC voted against moving the loose headers which are now in `include/` into `include/zephyr/`.
Therefore I can only presume the same considerations would apply to this change, as it would seem more disruptive.
So I label this as TSC: to be brought to the next TSC meeting and the attention of all TSC members.

nashif commented:
> The TSC voted against moving the loose headers which are now in `include/` into `include/zephyr/`.


We voted about phase 1 which is namespacing as mentioned in #12129: all zephyr headers will be included using a namespace, i.e. #include <zephyr/header.h>, #include <zephyr/subsystem/header.h>, etc.

mbolivar commented on include/misc/__assert.h:
```diff
@@ -1,129 +1,13 @@
- */
-
-#ifndef ZEPHYR_INCLUDE_MISC___ASSERT_H_
+#ifndef ZEPHYR_INCLUDE_MISC_RING_BUFFER_H_
```
??

nashif replied:
yeah, scripting foobar that i fixed at some point but missed to update  1 or 2 patches

mbolivar commented on include/misc/__assert.h:
```diff
@@ -1,129 +1,13 @@
+#include <sys/__assert.h>
 
-#endif /* ZEPHYR_INCLUDE_MISC___ASSERT_H_ */
+#endif /* ZEPHYR_INCLUDE_MISC_RING_BUFFER_H_ */
```
??

mbolivar commented on include/misc/byteorder.h:
```diff
@@ -1,328 +1,13 @@
+#include <sys/byteorder.h>
 
-#endif /* ZEPHYR_INCLUDE_MISC_BYTEORDER_H_ */
+#endif /* ZEPHYR_INCLUDE_MISC_RING_BUFFER_H_ */
```
Not sure what happened here (bad script?) but a lot of the include guards got messed up, it looks like.

mbolivar commented on include/misc/dlist.h:
```diff
@@ -1,554 +1,13 @@
- */
-
-#ifndef ZEPHYR_INCLUDE_MISC_DLIST_H_
+#ifndef ZEPHYR_INCLUDE_MISC_RING_BUFFER_H_
```
And here... you get the idea.

mbolivar commented on include/misc/mempool.h:
```diff
@@ -1,101 +1,13 @@
-	u32_t level : 4;
-	u32_t block : 28;
-};
+#warning "This header file has moved, include <sys/mempool.h> instead."
```
It would be nice to have a Kconfig to disable these warnings. Using the old include is the only way to write programs that are portable with this PR and LTS.

nashif replied:
yes, those are updated now.

mbolivar replied:
Unresolving; there still isn't a Kconfig option for this. Here's what I mean (I prefer option 2).

Option 1: use a bool

```
#ifndef CONFIG_V1_14_INCLUDES
#warning "This header file has moved, include <sys/__assert.h> instead."
#endif
```

That way, applications which need to target multiple Zephyr versions could add `CONFIG_V1_14_STYLE_INCLUDES=y` to their prj.conf.  And actually, if we want to be *really* nice to our users, this option can be `default y` at first, so users can turn it to `n` on their own time. After v1.15, it can be changed to `default n`.

Option 2: use a choice

```
choice
	prompt "Old-style include (v1.14 LTS compatible) support"
	default V1_14_INCLUDES_ALLOW

config V1_14_INCLUDES_ALLOW
	bool "Allow"
	help
	  Allow v1.14-style includes, like <misc/printk.h> instead of
	  <sys/printk.h>

config V1_14_INCLUDES_WARN
	bool "Warn"
	help
	  Emit a warning when v1.14-style includes are used.

config V1_14_INCLUDES_ERROR
	bool "Error"
	help
	  Emit a build error when v1.14-style includes are used.

endchoice
```

This is even better, and provides a way to manage this over multiple versions:

- v.1.15: default is `V1_14_INCLUDES_ALLOW`
- v1.16: default is `V1_14_INCLUDES_WARN`
- v1.17: default is `V1_14_INCLUDES_ERROR`
- v1.18: option is removed and all the shims go away

nashif replied:
I have seen this (and resolved by mistake), nice proposal, should be discussed before we go ahead and do it this way.

mbolivar commented on include/sys/dlist.h:
```diff
@@ -0,0 +1,554 @@
+ * simplifies the adding and removing of nodes to/from the list.
+ */
+
+#ifndef ZEPHYR_INCLUDE_MISC_DLIST_H_
```
Are you going to update these include guards too? (ZEPHYR_INCLUDE_SYS_DLIST_H_ etc.)

mbolivar commented on include/storage/flash_map.h:
```diff
@@ -0,0 +1,220 @@
+ * @brief Public API for flash map
+ */
+
+#ifndef ZEPHYR_INCLUDE_FLASH_MAP_H_
```
Looks like this commit also has issues with include guards.... does this actually work? It looks broken to me.

This file is included from include/flash_map.h, which has already defined ZEPHYR_INCLUDE_FLASH_MAP_H_.

nashif replied:
this is broken indeed, if you use the old location, in the tree however we removed usage of old location, so all works. Need to test allowing deprecation messages. Fixed now.

mbolivar replied:
Thanks

mbolivar commented on include/console/console.h:
```diff
@@ -0,0 +1,132 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_CONSOLE_H_
```
Same broken header guard updates in this patch as previous ones.

If we're going to go this route, I think there should be sanitycheck test cases that use the old style, so we know they still work. As said before, this is going to be really important if anyone wants to maintain an application that's compatible with LTS and this new hierarchy.

mbolivar commented on include/cmsis_rtos_v1/cmsis_os.h:
```diff
@@ -1,715 +1,13 @@
+
+#warning "This header file has moved, include <portable/cmsis_os.h> instead."
+
+#include <portable/cmsis_os.h>
```
I'm trying hard not to bikeshed the names here, but portable is particularly weird. Isn't almost every API somehow about portability? E.g. the periph/gpio.h API is for portability between different GPIO IP blocks.

This is just about CMSIS as far as I can tell; why is it not arm/cmsis/xxx.h or so instead?

nashif replied:
this is not really arm related, those APIs are a portability layer on top of Zephyr kernel using the CMSIS RTOS api abstraction. portable is just a placeholder here, bikeshedding is fine :)

mbolivar reviewed (commented):
Deliberately not commenting on whether or not to do this :).

Just have some concrete feedback on issues with what's been done.

pfalcon commented:
> all zephyr headers will be included using a namespace, i.e. #include <zephyr/header.h>, #include <zephyr/subsystem/header.h>, etc.

+100 to this great change, but where is it, I don't see it here?

nashif commented:
>> all zephyr headers will be included using a namespace, i.e. #include <zephyr/header.h>, #include <zephyr/subsystem/header.h>, etc.

> +100 to this great change, but where is it, I don't see it here?

This change proposal was dismissed.

pfalcon commented:
> This change proposal was dismissed.

Ah, but of course!

vanti commented on ext/hal/ti/simplelink/source/ti/devices/cc13x2_cc26x2/driverlib/uart.c:
```diff
@@ -36,7 +36,7 @@
 ******************************************************************************/
 
-#include "uart.h"
+#include <periph/uart.h>
```
This should not be changed. The file is referring to ./ext/hal/ti/simplelink/source/ti/devices/cc13x2_cc26x2/driverlib/uart.h, not the one in ./include

galak commented:
> * Moved all peripheral APIs into periph/

I'd ask we make this drivers/ instead.

galak commented:
Is there a script or something that can remove the shim headers?

galak commented:
kernel_version.h -> version.h?

galak commented:
toolchain.h -> toolchain/toolchain.h?

nashif commented:
> Is there a script or something that can remove the shim headers?

for i in `git grep -l "This header file has moved"`; do rm  $i; done

:)

galak commented:
tracing.h -> debug/tracing/tracing.h?

nashif commented:
> toolchain.h -> toolchain/toolchain.h?

yep, this actually should be private.

pfalcon commented:
> I'd ask we make this drivers/ instead.

But why? Peripherals and drivers are 2 different concepts, drivers just implement some internal functions which are consumed by periph subsystems (GPIO, I2C, etc.), and get exposed by periph subsystem public API. Mixing up internal drivers and public periph APIs (or more precisely, biasing that mix towards "drivers" instead of "public periph APIs") may get confusing.

galak commented:
It feels like we should have a `kernel` dir with things like `syscall.h`, `app_memory`, the stuff in `private`, maybe toss in `sys_clock.h`?

galak commented:
stats.h -> subsys/stats/stats.h

galak commented:
atomic.h & cache.h into arch/?

galak commented:
Do want to move linker/ under toolchain/ ?

joerchan reviewed (approved):
Bluetooth part looks good.

tbursztyka reviewed (commented):
instead of "include/periph" , wouldn't "include/devices" be better? After all, all the APIs there are dealing with struct device type

nvlsianpu commented:
^^
`include /periph` is coherent with the zephyr documentation (peripheral driver) and at last with usual MCU-world naming for MCU components like timers, serial controller etc.

nvlsianpu reviewed (approved):
I have been looking at changes related to: settings, flash_map, include/flash , soc_flash_nrf, flash simulator, NVS, FCB.

nashif commented:
> kernel_version.h -> version.h?

we already have version.h

`./build/zephyr/include/generated/version.h`

vanti reviewed (approved):
SimpleLink changes look good

jukkar reviewed (approved):
LGTM

ceolin commented on include/kernel.h:
```diff
@@ -14,7 +14,7 @@
 
 #if !defined(_ASMLANGUAGE)
-#include <kernel_includes.h>
+#include <private/kernel_includes.h>
```
do we intent to work on it in future ? It's really weird include private headers in a public header.

nashif replied:
that is  actually a lie, kernel_includes.h is not private, it just includes many other headers and should not be included directly. This will be changed.

lemrey reviewed (approved):
I understand @pfalcon's argument, but like @galak I would also rather see `drivers/` than `periph/` because it matches the tree structure.

nashif committed:
cleanup: include/: move flash_map.h to storage/flash_map.h

move flash_map.h to storage/flash_map.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move console.h to console/console.h

move console.h to console/console.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move tty.h to console/tty.h

move tty.h to console/tty.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move fs.h to fs/fs.h

move fs.h to fs/fs.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move fcb.h to fs/fcb.h

move fcb.h to fs/fcb.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move nvs/nvs.h to fs/nvs.h

move nvs/nvs.h to fs/nvs.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move tracing.h to debug/tracing.h

move tracing.h to debug/tracing.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move stats.h to stats/stats.h

move stats.h to stats/stats.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move atomic.h to sys/atomic.h

move atomic.h to sys/atomic.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move base64.h to sys/base64.h

move base64.h to sys/base64.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move crc.h to sys/crc.h

move crc.h to sys/crc.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move disk_access.h to disk/disk_access.h

move disk_access.h to disk/disk_access.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move power.h to power/power.h

move power.h to power/power.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move sys_io.h to sys/sys_io.h

move sys_io.h to sys/sys_io.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move json.h to data/json.h

move json.h to data/json.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move zephyr/jwt.h to data/jwt.h

move zephyr/jwt.h to data/jwt.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move watchdog.h to drivers/watchdog.h

move watchdog.h to drivers/watchdog.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move adc.h to drivers/adc.h

move adc.h to drivers/adc.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move can.h to drivers/can.h

move can.h to drivers/can.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move aio_comparator.h to drivers/aio_comparator.h

move aio_comparator.h to drivers/aio_comparator.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move clock_control.h to drivers/clock_control.h

move clock_control.h to drivers/clock_control.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move dma.h to drivers/dma.h

move dma.h to drivers/dma.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move counter.h to drivers/counter.h

move counter.h to drivers/counter.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move entropy.h to drivers/entropy.h

move entropy.h to drivers/entropy.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move flash.h to drivers/flash.h

move flash.h to drivers/flash.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move gna.h to drivers/gna.h

move gna.h to drivers/gna.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move gpio.h to drivers/gpio.h

move gpio.h to drivers/gpio.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move hwinfo.h to drivers/hwinfo.h

move hwinfo.h to drivers/hwinfo.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move i2c.h to drivers/i2c.h

move i2c.h to drivers/i2c.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move i2s.h to drivers/i2s.h

move i2s.h to drivers/i2s.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move ipm.h to drivers/ipm.h

move ipm.h to drivers/ipm.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move led.h to drivers/led.h

move led.h to drivers/led.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move led_strip.h to drivers/led_strip.h

move led_strip.h to drivers/led_strip.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move pinmux.h to drivers/pinmux.h

move pinmux.h to drivers/pinmux.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move pwm.h to drivers/pwm.h

move pwm.h to drivers/pwm.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move rtc.h to drivers/rtc.h

move rtc.h to drivers/rtc.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move sensor.h to drivers/sensor.h

move sensor.h to drivers/sensor.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move spi.h to drivers/spi.h

move spi.h to drivers/spi.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move uart.h to drivers/uart.h

move uart.h to drivers/uart.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move display.h to drivers/display.h

move display.h to drivers/display.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move ring_buffer.h to sys/ring_buffer.h

move ring_buffer.h to sys/ring_buffer.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
include/: cleanup: add Kconfig for suppressing warnings

Suppress warnings from deprecated header shims. Use when you have an
application that needs to work with older versions of Zephyr where
headers were located in different directories.

The shim will go away after two releases, so make sure to adapt your
application for the new locations.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/__assert.h to sys/__assert.h

move misc/__assert.h to sys/__assert.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/dlist.h to sys/dlist.h

move misc/dlist.h to sys/dlist.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/byteorder.h to sys/byteorder.h

move misc/byteorder.h to sys/byteorder.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/errno_private.h to sys/errno_private.h

move misc/errno_private.h to sys/errno_private.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/fdtable.h to sys/fdtable.h

move misc/fdtable.h to sys/fdtable.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/libc-hooks.h to sys/libc-hooks.h

move misc/libc-hooks.h to sys/libc-hooks.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/math_extras.h to sys/math_extras.h

move misc/math_extras.h to sys/math_extras.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/list_gen.h to sys/list_gen.h

move misc/list_gen.h to sys/list_gen.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/math_extras_impl.h to sys/math_extras_impl.h

move misc/math_extras_impl.h to sys/math_extras_impl.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/mempool_base.h to sys/mempool_base.h

move misc/mempool_base.h to sys/mempool_base.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/mempool.h to sys/mempool.h

move misc/mempool.h to sys/mempool.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/mutex.h to sys/mutex.h

move misc/mutex.h to sys/mutex.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/printk.h to sys/printk.h

move misc/printk.h to sys/printk.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/rb.h to sys/rb.h

move misc/rb.h to sys/rb.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/sflist.h to sys/sflist.h

move misc/sflist.h to sys/sflist.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/slist.h to sys/slist.h

move misc/slist.h to sys/slist.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/speculation.h to sys/speculation.h

move misc/speculation.h to sys/speculation.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/util.h to sys/util.h

move misc/util.h to sys/util.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/gcov.h to debug/gcov.h

move misc/gcov.h to debug/gcov.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/stack.h to debug/stack.h

move misc/stack.h to debug/stack.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
cleanup: include/: move misc/reboot.h to power/reboot.h

move misc/reboot.h to power/reboot.h and
create a shim for backward-compatibility.

No functional changes to the headers.
A warning in the shim can be controlled with CONFIG_COMPAT_INCLUDES.

Related to #16539

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
CODEOWNERS: update file with new locations

Adapt file to the new location of headers files in the tree.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
docs: make doxygen happy with new locations

Change @file entry on some headers due to duplicated files in the
include/ tree (real header + shim).
Adapt the doxygen configuration file to the new locations and remove
directories which are already included.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

carlescufi reviewed (approved):
Everything looks as agreed, nice job.

ioannisg reviewed (approved):
Thanks for the cleanup
